### ISSUE TITLE
mmp error MM0000: Unexpected error on XM with 2018-04

### ISSUE THREAD

**Issue by alexischr** [mono/mono#8634]
## Steps to Reproduce

1. Install a system Mono before version 5.14 (e.g 5.12) 
2. Install a package from https://github.com/xamarin/xamarin-macios/pull/3856, e.g.  https://bosstoragemirror.azureedge.net/wrench/macios-mac-mono-2018-04/b5/b5f79ca4d27d567245c634077c1dd2033060aaa3/xamarin.mac-4.5.0.361.pkg
3. Run `mmp --version`

### Current Behavior

```
error MM0000: Unexpected error - Please file a bug report at http://bugzilla.xamarin.com
System.MissingMethodException: Method 'string.Split' not found.
  at Mono.Options.OptionSet+ActionOption..ctor (System.String prototype, System.String description, System.Int32 count, System.Action`1[T] action, System.Boolean hidden) [0x00000] in /Users/builder/data/lanes/6026/b5f79ca4/source/xamarin-macios/external/mono/mcs/class/Mono.Options/Mono.Options/Options.cs:903
  at Mono.Options.OptionSet.Add (System.String prototype, System.String description, System.Action`1[T] action, System.Boolean hidden) [0x00026] in /Users/builder/data/lanes/6026/b5f79ca4/source/xamarin-macios/external/mono/mcs/class/Mono.Options/Mono.Options/Options.cs:930
  at Mono.Options.OptionSet.Add (System.String prototype, System.String description, System.Action`1[T] action) [0x00001] in /Users/builder/data/lanes/6026/b5f79ca4/source/xamarin-macios/external/mono/mcs/class/Mono.Options/Mono.Options/Options.cs:923
  at Xamarin.Bundler.Driver.Main2 (System.String[] args) [0x00001] in /Users/builder/data/lanes/6026/b5f79ca4/source/xamarin-macios/tools/mmp/driver.cs:233
  at Xamarin.Bundler.Driver.Main (System.String[] args) [0x00015] in /Users/builder/data/lanes/6026/b5f79ca4/source/xamarin-macios/tools/mmp/driver.cs:220
```

### Expected Behavior
```
mmp 4.5.0.361 (master: b5f79ca4)
```

## On which platforms did you notice this

[X] macOS
[ ] Linux
[ ] Windows

**Comment by lewurm** [comment]
@marek-safar could this be caused by the corefx string import?

**Comment by marek-safar** [comment]
This is mmp issue it needs to be built against the minimal API is suppose to run on.

@chamons where is the build script for mmp

**Comment by chamons** [comment]
- https://github.com/xamarin/xamarin-macios/blob/master/tools/mmp/Makefile
- https://github.com/xamarin/xamarin-macios/blob/master/tools/mmp/mmp.csproj

**Comment by marek-safar** [comment]
@chamons Yep, they produce very different output. Could we unify this and have only 1 build script for same output, which one do you prefer?

**Comment by chamons** [comment]
The makefile is the "real" one. The csproj is just for debugging.

It wouldn't be a bad idea to unify them at some point.

**Comment by marek-safar** [comment]
@chamons it'd be easier to unify by using csproj only, any objections?

**Comment by chamons** [comment]
The only objection is that Make is significantly faster than msbuild to build the projects.

However....I think updating mmp and mtouch to use the csproj and having the makefile msbuild /nologo /v:normal it is fine.

**Comment by marek-safar** [comment]
Fixed in xamarin-macios@b3bed1a8b439bda8f144b917592e805815af41d9
